src/Analyzers/shared/FeatureDetection/ProjectFeatureDetector.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.VisualStudio.LanguageServices;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Threading;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.AspNetCore.Analyzers.FeatureDetection
+{
+    // Be very careful making changes to this file. No project in this repo builds it.
+    //
+    // If you need to verify a change, make a local project (net472) and copy in everything included by this project.
+    //
+    // You'll also need some nuget packages like:
+    // - Microsoft.VisualStudio.LanguageServices
+    // - Microsoft.VisualStudio.Shell.15.0
+    // - Microsoft.VisualStudio.Threading
+    [Export(typeof(ProjectFeatureDetector))]
+    internal class ProjectFeatureDetector
```

Comment:
>ProjectFeatureDetector [](start = 19, length = 22)

Small ask, should we rename it to ProjectCompillationFeatureDetector? Or you plan to use it for any other types of features?

Reply:
Let me know please where i could get a source package when this is merged.

---
In reply to: [282145188](https://github.com/aspnet/AspNetCore/pull/10065#discussion_r282145188) [](ancestors = 282145188)

Reply:
Yeah, I'm happy to change the name 👍